### PR TITLE
Set retired questions to Unspecified if not answered

### PR DESCRIPTION
**Pull request by squirrelo**
This takes care of pulldowns that have a mix of answered retired question and surveys where the question was retired. It simply makes the retired question Unspecified.

### CODE REVIEW

**Comment by josenavas** [comment]
:+1: @wasade @mortonjt wanna review/merge?

File: knimin/lib/data_access.py [hunk 1]
```diff
@@ -799,9 +799,14 @@ def pulldown(self, barcodes, blanks):
             survey_md = [''.join(['sample_name\t', '\t'.join(headers)])]
             for barcode, shortnames_answers in sorted(bc_responses.items()):
                 barcodes_seen.add(barcode)
-                ordered_answers = [shortnames_answers[h] for h in headers]
                 oa_hold = [barcode]
-                for x in ordered_answers:
+                for h in headers:
+                    # Take care of retired questions not having an answer
+                    try:
```
Comment (wasade) [inline knimin/lib/data_access.py]: `x` is a bad variable name

why not `answer = shortnames_answers.get(h, 'Unspecified')`?

**Comment by wasade** [comment]
tests are failing as well -- is that expected?

**Comment by mortonjt** [comment]
I just tested the metadata pulldown.  It seems to work as expected.

File: knimin/lib/data_access.py (hunk 1 above)
Comment (squirrelo) [inline knimin/lib/data_access.py]: Done

**Comment by squirrelo** [comment]
tests failing is expected: flake8 is picking up complex functions, and the test have not been updated to the new AG database pulldown.

**Comment by squirrelo** [comment]
Ready to go.

**Pull request merged**